### ISSUE TITLE
Simple auth formatting

### ISSUE THREAD

**Issue by AdamNuclear** [hashicorp/vault-secrets-operator#478]
Issue: Various projects are consuming secrets in a specific format
Example: htpasswd in secret - any webserver is expecting:

```
apiVersion: v1
data:
  users: <b64 encoded key:value  list of users>
```

decoded b64:
```
user:htpassword
```

Would be good to have formatting options like: 
```
kind: VaultStaticSecret
metadata:
  name: vault-kv-app
  namespace: app
spec:
  format: <string> | default (json) json/yaml/kv
  destination_key_name: <string> | default: data
```

optionally would be also great to have the possibility to select just keys from kv:

```
spec:
 keys:
 - 1secretkey
 - 2secretkey
```

This would make the secrets operator way more compatible with all the software around without rewriting/modifying other projects.

**Comment by adrianmoisey** [comment]
Would this issue be solved by https://github.com/hashicorp/vault-secrets-operator/pull/437 ?

**Comment by AdamNuclear** [comment]
yea that would definitely solve it!

**Comment by benashz** [comment]
Closed with #437
